rutil/Data.hxx

```diff
@@ -584,6 +607,18 @@ class Data
       */
       const char* c_str() const;
 
+      /**
+        Convert to a C++ string.
+      */
+      std::string toString() const;
```

Comment:
Hi @portsip,
let's finish our discussion regarding `operator std::sting() const` - https://github.com/resiprocate/resiprocate/pull/382#discussion_r1817399929. It seems we didn't understand each other.
I suggest adding an operator for implicit conversion.
```
operator std::string() const {
        return std::string(c_str(), size());
    }
```
It could be used in the following way `std::string val = resip::Data("test message");`
As a result, resip::Data will have two methods to convert to std::string - explicit `toString()` and implicit `operator std::string()`.
What do you think about this change?
The same question for std::string_view.